jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -387,48 +428,91 @@ V poll()
                 return null;
             }
         }
-
+        
         /**
-         * For every bucket B in this {@code BucketList}, which contains vertices from the set
-         * {@code
-         * vertices}, creates a new {@code Bucket} B' and moves vertices from B to B' according to
-         * the following rule: $B' = B\cap vertices$ and $B = B\backslash B'$. For every such
-         * {@code Bucket} B only one {@code Bucket} B' is created. If some bucket B becomes empty
+         * Split every bucket $B$ into buckets $B \cap N(vertex)$ and $B \setminus N(vertex)$. For every such
+         * {@code Bucket} B only one new {@code Bucket} B' is created. If some bucket B becomes empty
          * after this operation, it is removed from the data structure.
          *
-         * @param vertices the vertices, that should be moved to new buckets.
+         * @param vertex The vertex whose neighborhood is used for splitting.
          */
-        void updateBuckets(Set<V> vertices)
-        {
-            Set<Bucket> visitedBuckets = new HashSet<>();
-            for (V vertex : vertices) {
-                Bucket bucket = bucketMap.get(vertex);
-                if (visitedBuckets.contains(bucket)) {
-                    bucket.prev.addVertex(vertex);
-                    bucketMap.put(vertex, bucket.prev);
-                } else {
-                    visitedBuckets.add(bucket);
-                    Bucket newBucket;
-                    if (priorityB != null) {
-                        newBucket = new Bucket(vertex, new PriorityComparator(priorityA), new PriorityComparator(priorityB));
-                    }
-                    else{
-                        newBucket = new Bucket(vertex, priorityComparator);
+        void updateBuckets(V vertex) {
+            HashSet<Bucket> bucketsToUpdate = new HashSet<>();
+            Map<Bucket, List<V>> intersectionLists = new HashMap<Bucket, List<V>>();
+            Map<Bucket, List<V>> intersectionListsB = null;
+            
+            List<V> sortedNeighborhood = sortedNeighbors.get(vertex);
```

Comment:
You have sortedNeighborhood and sortedNeighbors. Bad naming style.

Reply:
I renamed the variables to `sortedNeighborsOfVertex`/`sortedNeighborsOfVertexB` -- do you agree with this?